algebra/matrix.c

```diff
@@ -202,7 +202,7 @@ c_float quad_form(const csc *P, const c_float *x) {
 #ifdef PRINTING
         c_eprint("quad_form matrix is not upper triangular");
 #endif /* ifdef PRINTING */
-        return OSQP_NULL;
+        return -1.;
```

Comment:
`OSQP_NULL` is defined as `0`, which is a valid result of `quad_form` function (e.g. when `P` is zero). I changed it to `-1` which makes it easier to detect the failure.